dali/pipeline/data/buffer.h

```diff
@@ -270,6 +271,21 @@ class Buffer {
 
   DISABLE_COPY_MOVE_ASSIGN(Buffer);
 
+  static void SetGrowthFactor(double factor) {
+    assert(factor >= 1.0);
+    growth_factor_ = factor;
```

Comment:
mutex lock?

Reply:
I think it's not necessary and definitely quite costly. We could make it atomic, but that's still a bit of overdoing things. It's not a critical resource and the function is extremely unlikely to be used at the time the value is used. The timing is not critical and there can be no race condition, since the value is only used once in `ResizeHelper`.

Reply:
fair enough